src/tracks/playabletrack/wavetrack/ui/WaveTrackView.cpp

```diff
@@ -1042,6 +1047,13 @@ void WaveTrackView::DoSetMinimized( bool minimized )
    } );
 }
 
+std::shared_ptr<CommonTrackCell> WaveTrackView::DoGetAffordance(const std::shared_ptr<Track>& track)
+{
+    if (mpAffordanceCellControl == nullptr)
+        mpAffordanceCellControl = std::make_shared<WaveTrackAffordanceControls>(track);
+    return mpAffordanceCellControl;
```

Comment:
I see a minor consequence, that if a track ever needs an affordance, but later no longer needs it (as when swapping stereo channels) -- then the affordance control is not deleted until the track is.  Not very important.